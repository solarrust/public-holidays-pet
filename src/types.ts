export interface RawCountry {
  id: string
  isoCode: string
  name: { language: string; text: string }[]
  officialLanguages: string[]
}

export interface Country {
  id: string
  isoCode: string
  name: string
  officialLanguages: string[]
}

export interface Holiday {
  comment: [
    {
      language: string
      text: string
    },
  ]
  endDate: Date
  id: string
  name: {
    language: string
    text: string
  }[]
  nationwide: boolean
  regionalScope: string
  startDate: Date
  subdivisions: [
    {
      code: string
      shortName: string
    },
  ]
  temporalScope: string
  type: string
}
