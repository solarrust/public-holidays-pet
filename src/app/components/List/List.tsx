import React from 'react';
import { format } from 'date-fns';

import { Holiday } from '@/types';

interface ListProps {
  holidays: Holiday[];
  placeholder: string;
}

export default function List({ holidays, placeholder }: ListProps) {
  return (
    <ul className="grid gap-2 mt-8 text-2xl">
      {holidays.length === 0 ? (
        <li>{placeholder}</li>
      ) : (
        holidays.map((item) => (
          <li key={item.id}>
            <span className="font-bold">{format(item.startDate, 'd MMMM yyyy, EEEE')}</span> â€” {item.name}
          </li>
        ))
      )}
    </ul>
  );
}
