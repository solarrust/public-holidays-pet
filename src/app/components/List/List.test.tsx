import React from 'react';

import { Holiday } from '@/types';
import { render } from '@testing-library/react';

import List from './List';

describe('List component', () => {
  const holidays: Holiday[] = [
    {
      id: '1',
      name: 'New Year',
      startDate: new Date('2023-01-01').toISOString(),
      endDate: new Date('2023-01-01').toISOString(),
      nationwide: false,
    },
    {
      id: '2',
      name: 'Christmas',
      startDate: new Date('2023-12-25').toISOString(),
      endDate: new Date('2023-12-25').toISOString(),
      nationwide: false,
    },
  ];

  it('renders holidays list correctly', () => {
    const { getByText } = render(<List holidays={holidays} placeholder="No holidays available" />);
    expect(getByText('1 January 2023, Sunday')).toBeInTheDocument();
    expect(getByText('25 December 2023, Monday')).toBeInTheDocument();
  });

  it('matches snapshot when holidays list is empty', () => {
    const { asFragment } = render(<List holidays={[]} placeholder="No holidays available" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
