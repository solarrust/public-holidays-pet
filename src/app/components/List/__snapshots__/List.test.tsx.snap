// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component matches snapshot when holidays list is empty 1`] = `
<DocumentFragment>
  <ul
    class="grid gap-2 mt-8 text-2xl"
  >
    <li>
      No holidays available
    </li>
  </ul>
</DocumentFragment>
`;

exports[`List component matches snapshot when holidays list is populated 1`] = `
<DocumentFragment>
  <ul
    class="grid gap-2 mt-8 text-2xl"
  >
    <li>
      <span
        class="font-bold"
      >
        1 January 2023, Sunday
      </span>
       — New Year
    </li>
    <li>
      <span
        class="font-bold"
      >
        25 December 2023, Monday
      </span>
       — Christmas
    </li>
  </ul>
</DocumentFragment>
`;
